doc/tutorials/image/image.txt

```diff
@@ -0,0 +1,185 @@
+
+@section imageinfo_api_imagetut ImageInfo
+
+ImageInfo class contains the metadata of the images.
+@code
+  /**
+   * Instantiate the ImageInfo object with the image width, height, channels.
+   *
+   * @param width Image width.
+   * @param height Image height.
+   * @param channels number of channels in the image.
+   */
+  ImageInfo(const size_t width,
+            const size_t height,
+            const size_t channels);
+@endcode
+Other public memebers include:
+  - flipVertical Flip the image vertical upon loading.
+  - quality Compression of the image if saved as jpg (0-100).
+
+@section load_api_imagetut Load
+
+
+Standalone loading of images.
+@code
+  /**
+   * Load the image file into the given matrix.
+   *
+   * @param filename Name of the image file.
+   * @param matrix Matrix to load the image into.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Load(const std::string& filename,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+Loading a test image. It also fills up the ImageInfo class object.
+@code
+data::ImageInfo info;
+data::Load("test_image.png", matrix, info, false, true);
+@endcode
+
+If the size of image is known beforehand, it can be passed into the constructor of ImageInfo to set the size of outputMatrix(the matrix into which the image is loaded).
+
+@code
+size_t height = 64, width = 64, channels = 1;
+data::ImageInfo info(width, height, channels);
+@endcode
+
+More than one image can be loaded into the same matrix.
+
+Loading multiple images:
+
+@code
+  /**
+   * Load the image file into the given matrix.
+   *
+   * @param files A vector consisting of filenames.
+   * @param matrix Matrix to save the image from.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Load(const std::vector<std::string>& files,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+@code
+  data::ImageInfo info;
+  std::vector<std::string>> files{"test_image1.bmp","test_image2.bmp"};
+  data::load(files, matrix, info, false, true);
+@endcode
+
+@section save_api_imagetut Save
+
+Save images expects a matrix of type unsigned char in the form (width * height * channels, n_images).

```

Comment:
``(width * height * channels, n_images)`` not sure about the ``n_images`` parameter, maybe ``NumberOfImages`` is more clear?

Reply:
okay, I'll change